command/job_periodic_force_test.go

```diff
@@ -0,0 +1,204 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/nomad/nomad/mock"
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+func TestJobPeriodicForceCommand_Implements(t *testing.T) {
+	t.Parallel()
+	var _ cli.Command = &JobPeriodicForceCommand{}
+}
+
+func TestJobPeriodicForceCommand_Fails(t *testing.T) {
+	t.Parallel()
+	ui := new(cli.MockUi)
+	cmd := &JobPeriodicForceCommand{Meta: Meta{Ui: ui}}
+
+	// Fails on misuse
+	if code := cmd.Run([]string{"some", "bad", "args"}); code != 1 {
+		t.Fatalf("expected exit code 1, got: %d", code)
+	}
+	if out := ui.ErrorWriter.String(); !strings.Contains(out, commandErrorText(cmd)) {
+		t.Fatalf("expected help output, got: %s", out)
+	}
+	ui.ErrorWriter.Reset()
+
+	if code := cmd.Run([]string{"-address=nope", "12"}); code != 1 {
+		t.Fatalf("expected exit code 1, got: %d", code)
+	}
+	if out := ui.ErrorWriter.String(); !strings.Contains(out, "Error forcing periodic job") {
+		t.Fatalf("expected failed to force error, got: %s", out)
+	}
+	ui.ErrorWriter.Reset()
+}
+
+func TestJobPeriodicForceCommand_AutocompleteArgs(t *testing.T) {
+	t.Parallel()
+
+	srv, _, url := testServer(t, true, nil)
+	defer srv.Shutdown()
+
+	ui := new(cli.MockUi)
+	cmd := &JobPeriodicForceCommand{Meta: Meta{Ui: ui, flagAddress: url}}
+
+	// Create a fake job, not periodic
+	state := srv.Agent.Server().State()
+	j := mock.Job()
+	require.NoError(t, state.UpsertJob(1000, j))
+
+	predictor := cmd.AutocompleteArgs()
+
+	res := predictor.Predict(complete.Args{Last: j.ID[:len(j.ID)-5]})
+	require.Empty(t, res)
+
+	// Create another fake job, periodic
+	state = srv.Agent.Server().State()
+	j2 := mock.Job()
+	j2.Periodic = &structs.PeriodicConfig{
+		Enabled:         true,
+		Spec:            "spec",
+		SpecType:        "cron",
+		ProhibitOverlap: true,
+		TimeZone:        "test zone",
+	}
+	require.NoError(t, state.UpsertJob(1000, j2))
+
+	res = predictor.Predict(complete.Args{Last: j2.ID[:len(j.ID)-5]})
+	require.Equal(t, []string{j2.ID}, res)
+
+	res = predictor.Predict(complete.Args{})
+	require.Equal(t, []string{j2.ID}, res)
+}
+
+func TestJobPeriodicForceCommand_NonPeriodicJob(t *testing.T) {
+	t.Parallel()
+	srv, client, url := testServer(t, true, nil)
+	defer srv.Shutdown()
+	testutil.WaitForResult(func() (bool, error) {
+		nodes, _, err := client.Nodes().List(nil)
+		if err != nil {
+			return false, err
+		}
+		if len(nodes) == 0 {
+			return false, fmt.Errorf("missing node")
+		}
+		if _, ok := nodes[0].Drivers["mock_driver"]; !ok {
+			return false, fmt.Errorf("mock_driver not ready")
+		}
+		return true, nil
+	}, func(err error) {
+		t.Fatalf("err: %s", err)
```

Comment:
The pattern we are adopting elsewhere is to use `require.Fail` or `require.NoError` - it makes the error more obvious when looking at stream of log messages.